src/actions/actions.ts

```diff
@@ -611,6 +611,14 @@ class CommandInsertRegisterContent extends BaseCommand {
 
     if (register.text instanceof Array) {
        text = (register.text as string []).join("\n");
+    } else if (register.text instanceof RecordedState) {
+      vimState.recordedState.transformations.push({
```

Comment:
If we're actually performing a macro at this point, then we absolutely should use a transformation.

But it looked to me like we were just storing the content of the macro in a register, which doesn't require any text transformations.